src/query/storage/m3/cluster_resolver.go

```diff
@@ -119,6 +121,9 @@ func resolveClusterNamespacesForQuery(
 	// Filter aggregated namespaces by filter function and options.
 	var r reusedAggregatedNamespaceSlices
```

Comment:
nit: Maybe we should rename `reusedAggregatedNamespaceSlices` to `reusedNamespaceSlices` since sometimes we put the unaggregated namespace in here now?